### ISSUE TITLE
Failed to activate the proto-repl package

### ISSUE THREAD

**Issue by robahl** [jasongilman/proto-repl#156]
After upgrading to proto-repl version `1.4.7`, Atom version `1.9.9`.

**Description**
Autocomplete provider 'Object(undefined)'
specifies `selector` instead of the `scopeSelector` attribute.
See https://github.com/atom/autocomplete-plus/wiki/Provider-API.

**Stacktrace**

> Error: Autocomplete provider 'Object(undefined)'
> specifies `selector` instead of the `scopeSelector` attribute.
> See https://github.com/atom/autocomplete-plus/wiki/Provider-API.
>     at ProviderManager.module.exports.ProviderManager.registerProvider (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/provider-manager.js:302:17)
>     at ProviderManager.registerProvider (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/provider-manager.js:3:61)
>     at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/main.js:83:65
>     at Object.module.exports.requireAutocompleteManagerAsync (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/main.js:23:49)
>     at Object.module.exports.consumeProvider (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/main.js:79:12)
>     at Object.module.exports.consumeProvider_3_0 (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/autocomplete-plus/lib/main.js:65:19)
>     at Provider.module.exports.Provider.provide (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/provider.js:30:52)
>     at ServiceHub.module.exports.ServiceHub.provide (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/service-hub.js:30:20)
>     at Package.module.exports.Package.activateServices (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:352:71)
>     at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:186:16)
>     at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:156:32
>     at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
>     at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:149:26
>     at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:146:34)
>     at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:538:21)
>     at /Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:342:19
>     at /Applications/Atom.app/Contents/Resources/app.asar/src/config.js:631:20
>     at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
>     at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
>     at Config.module.exports.Config.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:816:29)
>     at Config.module.exports.Config.setRawValue (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:608:19)
>     at Config.module.exports.Config.set (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:227:14)
>     at Config.module.exports.Config.removeAtKeyPath (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:377:12)
>     at Package.module.exports.Package.enable (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:77:26)
>     at PackageManager.module.exports.PackageManager.enablePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:169:14)
>     at HTMLButtonElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-card.js:324:27)
>     at HTMLButtonElement.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
>     at HTMLButtonElement.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)

**Comment by jasongilman** [comment]
I can reproduce this as well but just during the update. If you restart Atom the problem will go away. I think this is related to #85. The code in Proto REPL isn't reloaded after an update. I need to see why that's happening.

**Comment by robahl** [comment]
Thanks! Restarting Atom worked!

**Comment by pythys** [comment]
As help for anyone who's suffering from this bug given that I still experience it in the latest version, what I did is to modify the coffeescript **$HOME/.atom/packages/proto-repl/lib/process/lein-runner.coffee** as show in below diff

`@@ -19,7 +19,7 @@ module.exports = (currentWorkingDir, leinPath, args) ->`
`filteredEnv["Path"] = envPath + path.delimiter + leinPath`
`else`
`# Mac/Linux`
`-      leinExec = "lein"`
`+      leinExec = "/opt/leiningen/lein"`
`envPath = filteredEnv["PATH"] || ""`
`filteredEnv["PATH"] = envPath + path.delimiter + leinPath`

In other words, I hardcoded the full path of where leiningen resides. This should fix the issue temporarily. Make sure you choose whatever fits your OS
